core/src/main/java/de/danoeh/antennapod/core/service/download/DownloadService.java

```diff
@@ -335,8 +335,11 @@ private void handleFailedDownload(Downloader downloader) {
                             && String.valueOf(HttpURLConnection.HTTP_GONE).equals(status.getReasonDetailed());
                     boolean httpBadReq = status.getReason() == DownloadError.ERROR_HTTP_DATA_ERROR
                             && String.valueOf(HttpURLConnection.HTTP_BAD_REQUEST).equals(status.getReasonDetailed());
+                    boolean ioError = status.getReason() == DownloadError.ERROR_IO_ERROR;
+                    boolean connectError = status.getReason() == DownloadError.ERROR_CONNECTION_ERROR;
 
-                    if (httpNotFound || forbidden || notEnoughSpace || wrongFileType || httpGone || httpBadReq ) {
+                    if (httpNotFound || forbidden || notEnoughSpace || wrongFileType
+                            || httpGone || httpBadReq || ioError || connectError) {
```

Comment:
This now checks nearly all errors that can occur. How about explicitly listing cases where we want it to retry? I think there are not many cases where we want that, as the many added conditions in the last years show :)

Reply:
Yeah, good point.  It will be hard for me to test all the possible scenarios, I'll give it a try

Reply:
ready for a review @ByteHamster